src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -389,6 +390,20 @@ package object metadata extends DebugEnhancedLogging {
     }
   }
 
+  def filesXmlFilesHaveOnlyAllowedAccessRights(t: TargetBag): Try[Unit] = {
+    trace(())
+      for {
+        xml <- t.tryFilesXml
+        _ <- (xml \ "file" \ "accessRights").map(validateAccessRights).collectResults.recoverWith {
+          case ce: CompositeException => Try(fail(ce.getMessage))
+        }
+      } yield ()
+    }
+
+  private def validateAccessRights(accessRights: NodeSeq): Try[Unit] = Try {
```

Comment:
It would be great if the error message you generate here would also point to which file has an invalid access right. You can find that in the `filepath` attribute. So maybe pass along the whole `<file/>` node to this method.

(Side note: I think it should be `Node` rather than `NodeSeq` here. Then you also don't have to use that `foreach` construction below.)

Reply:
I have looked into getting the `file` in the error message, but it made the code that much more complex that I think it is not worth it.  If you @rvanheest have a neat solution, I would be interested to see it:)